### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by serentinos**
Some tests are not passed, but it works, I've checked :D

### CODE REVIEW

**Review by 33Orange — CHANGES_REQUESTED** [review]
Almost done

Please fix all errors and try to pass the tests.

File: src/services/expenses.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+let expenses = [];
+
+const getAll = () => {
+  return expenses;
+};
+
+const getById = (expenseId) => {
+  const foundUser = expenses.find(expense => expense.id === +expenseId);
+
+  return foundUser || null;
```
[the remark this answers is not in the document]
Comment (serentinos) [reply src/services/expenses.js:14]: typo

File: src/services/expenses.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    if (to) {
+      const dateTo = new Date(to);
+      const currentDate = new Date(expense.spentAt);
+
+      isMatch = currentDate <= dateTo;
+    }
+
+    return isMatch;
```
Comment (33Orange) [inline src/services/expenses.js:47]:
```suggestion
    if (userId) {
      return expense.userId === +userId;
    }

    if (category) {
       return category.includes(expense.category);
    }

    if (from) {
      const dateFrom = new Date(from);
      const currentDate = new Date(expense.spentAt);

      return currentDate >= dateFrom;
    }

    if (to) {
      const dateTo = new Date(to);
      const currentDate = new Date(expense.spentAt);

      return currentDate <= dateTo;
    }

    return true;
```

**Review by 33Orange — CHANGES_REQUESTED** [review]
Tests are failed.

image

**Pull request closed without merging**